stdlib/public/Concurrency/AsyncFlatMapSequence.swift

```diff
@@ -0,0 +1,95 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import Swift
+
+extension AsyncSequence {
+  @inlinable
+  public __consuming func flatMap<SegmentOfResult: AsyncSequence>(
+    _ transform: @escaping (Element) async -> SegmentOfResult
+  ) -> AsyncFlatMapSequence<Self, SegmentOfResult> {
+    return AsyncFlatMapSequence(self, transform: transform)
+  }
```

Comment:
What do you think about having an overload like this? (I really think we need some an `AsyncAdapter`, like the one in your tests.)

```swift
func flatMap<SegmentOfResult: Sequence>(
    _ transform: @escaping (Element) async -> SegmentOfResult
) -> AsyncFlatMapSequence<Self, AsyncAdapter<SegmentOfResult>>
```

Reply:
yea, I need to make a pitch for that (currently working on that)